### PR TITLE
#139 - frontend/visual table of saved datasets/graphs

### LINKED ISSUE #139: [Frontend] - Visual Table of Saved Datasets/Graphs
As a User, I would like to visually see the saved datasets/graphs in a table/list.

**Description:**
This is the frontend portion of the saved data #14 Task. It will include the table with a basic layout, a table with correct headers, and a list of the datasets and graphs along with unique details for each one.

**Tasks Involved:**

- [ ]  Create basic layout for table
- [ ]  Create table with headers
- [ ]  Display datasets/graphs in comprehensive fashion (Details to distinguish between them all)

The mockup can be found in the task #14 

**Acceptance Criteria**
- Table/List exists
- List of Datasets/Graphs
- Unique details for each (name of dataset, unique graph name, etc)

**Story Point** - 1
**RIsk** - Low
**Priority** - Medium

### PR DESCRIPTION
**Pull request by laithAwad**
A profile page was created, and on that page a table that will display the saved datasets and graphs is created, the table has an expandable options to each entry inside it to show more information about the dataset or the graph in that row. I have also made the names of each row data as a link that will redirect the user to the graph page and send a query to that page containing the dataset ids of the table row selected. On GraphView page I have added an npm package called query-string that can parse the sent query into an object with the appropriate values, this may or may not be needed depending on the backend way of handling the queries.

### CODE REVIEW

**Comment by laithAwad** [comment]
Screen Shot 2020-12-15 at 9 14 07 PM

Screenshot showing how the table looks like

File: Client/src/Views/ProfileView.tsx [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+const useRowStyles = makeStyles({
+  root: {
+    '& > *': {
+      borderBottom: 'unset',
+    },
+  },
+})
+
+//no oxidizer, add material detials and composition, comments keep author
+function createData(id, type: string, name: string, oxidizer: string, author: string, graphdatasets, comments: string) {
```
Comment (GD-Dheer) [inline Client/src/Views/ProfileView.tsx]: Is graphdatasets in missing a type here?

File: Client/src/Views/ProfileView.tsx [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+}
+
+function createGraphData(id, type: string, name: string, graphdatasets, comments: string) {
+  return { id, type, name, graphdatasets, comments }
+}
+
+rows.push(createGraphData([1, 2, 3], 'Graph', 'graph name', ['Cell width', 'Cell Height of O2 explosion', 'Critical energy after N2 intake'],
+  'Here is a sample comment'))
+
+function parseIdArray(id) {
```
Comment (GD-Dheer) [inline Client/src/Views/ProfileView.tsx:68]: id: number

**Review by GD-Dheer — APPROVED** [review]
These changes look good. Saw a demo, looks like the mockup

File: Client/src/Views/ProfileView.tsx (hunk 1 above)
Comment (laithAwad) [reply Client/src/Views/ProfileView.tsx]: yes, i added it

File: Client/src/Views/ProfileView.tsx [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+  return (
+    < React.Fragment >
+      <TableRow className={classes.root}>
+        <TableCell>
+          <IconButton aria-label="expand row" size="small" onClick={() => setOpen(!open)}>
+            {open ? <KeyboardArrowUpIcon /> : <KeyboardArrowDownIcon />}
+          </IconButton>
+        </TableCell>
+        <TableCell component="th" scope="row">
+          <Link to={"graph?datasetId[]=" + String(parseIdArray(row.id))}>{row.name}</Link>
```
Comment (ds161833) [inline Client/src/Views/ProfileView.tsx:91]: why not just datasetIds=

**Review by ds161833 — APPROVED** [review]
Looks good although, the Profile View is a little confusing, I am having a hard time understanding what exactly the View renders, if you want we could take a look together, otherwise we can merge this PR

**Comment by JeremiahTee** [comment]
Looks like the "Expected — Waiting for status to be reported" is a problem with [Travis itself (related to migration) ](https://travis-ci.community/t/known-issue-travis-ci-reports-expected-waiting-for-status-to-be-reported-on-the-github-status-api-but-the-status-never-arrives/1154/31)

I suggest resolving the conflicts in this PR then pushing a commit. See if Travis passes both checks. @laithAwad

**Review by JeremiahTee — CHANGES_REQUESTED** [review]
No need to commit package-lock.json if I remember correctly it's a binary
Rest looks good, resolve conflicts then push the commits as mentioned
Also don't forget to check in with @ds161833 post break when he is back

File: Client/src/Views/ProfileView.tsx (hunk 3 above)
Comment (laithAwad) [reply Client/src/Views/ProfileView.tsx:91]: That was a request from Sean that he wants the string to be in this format

**Review by JeremiahTee — APPROVED** [review]
The changes requested were made. Great work, approved

**Pull request merged**